src/server/ua_services_attribute.c

```diff
@@ -1342,3 +1360,128 @@ __UA_Server_write(UA_Server *server, const UA_NodeId *nodeId,
     }
     return UA_Server_write(server, &wvalue);
 }
+
+#ifdef UA_ENABLE_HISTORIZING
+void
+Service_HistoryRead(UA_Server *server, UA_Session *session,
+                          const UA_HistoryReadRequest *request,
+                          UA_HistoryReadResponse *response) {
+    UA_HistoryReadResponse_init(response);
+    if (request->historyReadDetails.encoding != UA_EXTENSIONOBJECT_DECODED) {
+        response->responseHeader.serviceResult = UA_STATUSCODE_BADNOTSUPPORTED;
+        return;
+    }
+    if (request->historyReadDetails.content.decoded.type == &UA_TYPES[UA_TYPES_READRAWMODIFIEDDETAILS]) {
```

Comment:
Exit early with `UA_STATUSCODE_BADHISTORYOPERATIONUNSUPPORTED` if the type is unsupported.
This reduces the nesting level of the remaining function.